include/nuttx/syslog/syslog.h

```diff
@@ -243,6 +244,55 @@ int syslog_initialize(void);
 FAR struct syslog_channel_s *syslog_file_channel(FAR const char *devpath);
 #endif
 
+/****************************************************************************
+ * Name: syslog_stream_init
+ *
+ * Description:
+ *   Initialize to use the device stream as the SYSLOG sink.
+ *
+ *   On power up, the SYSLOG facility is non-existent or limited to very
+ *   low-level output.  This function may be called later in the
+ *   initialization sequence after full driver support has been initialized.
+ *   (via syslog_initialize())  It installs the configured SYSLOG drivers
+ *   and enables full SYSLOGing capability.
+ *
+ * Input Parameters:
+ *   stream - The stream device to be used.
+ *
+ * Returned Value:
+ *   Returns a newly created SYSLOG channel, or NULL in case of any failure.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SYSLOG_STREAM
+FAR struct syslog_channel_s *
+syslog_stream_init(FAR struct lib_outstream_s *stream);
+#endif
+
+/****************************************************************************
+ * Name: syslog_bknd_stream_uninit
+ *
+ * Description:
+ *   Disable the channel in preparation to use a different
+ *   SYSLOG device.
+ *
+ * Input Parameters:
+ *   channel    - Handle to syslog channel to be used.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned on
+ *   any failure.
+ *
+ * Assumptions:
+ *   The caller has already switched the SYSLOG source to some safe channel
+ *   (the default channel).
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SYSLOG_STREAM
+void syslog_stream_uninit(FAR struct syslog_channel_s *channel);
```

Comment:
remove? point by g_syslog_stream_ops

Reply:
Okay, I will remove it, since close will call it internally